security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/keycloak/SAMLUserDetailsServiceImpl.java

```diff
@@ -107,8 +107,7 @@ public Object loadUserBySAML(SAMLCredential credential)
             if (attrName.equals(SAML_IDP_METADATA_ROLE_ATTR_NAME)) {
                 List<XMLObject> attributeValues = cAttribute.getAttributeValues();
                 if (!attributeValues.isEmpty()) {
-                    userRoles.add(new StringBuilder(APP_NAME).append(":").append(
-                        getAttributeValue(attributeValues.get(0))).toString());
+                    userRoles.add(getAttributeValue(attributeValues.get(0)));
```

Comment:
I found the issue: when configuring the client in the Keycloak admin panel, the **Scope** tab seems to be set to ‘_Full scope allowed_’ by default. If I disable this, Keycloak only tells cBioPortal about the roles specifically defined for the cBioPortal client, and shows a panel to also explicitly include specific realm roles and other roles if required. If I add an instruction to flip this setting to our guide, we do not need any workarounds.